crates/sparrow-runtime/src/read/parquet_file.rs

```diff
@@ -107,7 +107,7 @@ impl ParquetFile {
 
         batch_stream = batch_stream.with_batch_size(batch_size.unwrap_or(BATCH_SIZE_ROWS));
         if let Some(projection) = projection {
-            let mask = ProjectionMask::leaves(
+            let mask = ProjectionMask::roots(
```

Comment:
`roots` pulls direct children, `leaves` unnests, so for `map` types we were getting the incorrect top-level schema.